packages/plugins/plugin-table/src/table-model.ts

```diff
@@ -70,15 +85,51 @@ export class TableModel extends Resource {
       const fields = this.table.view?.fields ?? [];
       return Object.fromEntries(
         fields.map((field, index: number) => {
-          return [getCellKey(index, 0), { value: field.label ?? field.path, resizeHandle: 'col' }];
+          return [
+            getCellKey(index, 0),
+            {
+              value: field.label ?? field.path,
+              resizeHandle: 'col',
+              accessoryHtml: columnSettingsButtonHtml(field.id),
+            },
+          ];
         }),
       );
     });
 
+    const sortedData = computed(() => {
+      this.displayToDataIndex.clear();
+      const sort = this.sorting.value;
+      if (!sort) {
+        return this.data;
+      }
+
+      const field = this.table.view?.fields.find((f) => f.id === sort.columnId);
+      if (!field) {
+        return this.data;
+      }
+
+      const dataWithIndices = this.data.map((item, index) => ({ item, index }));
+      const sorted = sortBy(dataWithIndices, [(wrapper) => wrapper.item[field.path]]);
+
+      if (sort.direction === 'desc') {
+        sorted.reverse();
+      }
+
+      for (let displayIndex = 0; displayIndex < sorted.length; displayIndex++) {
+        const { index: dataIndex } = sorted[displayIndex];
+        if (displayIndex !== dataIndex) {
+          this.displayToDataIndex.set(displayIndex, dataIndex);
+        }
+      }
+
+      return sorted.map(({ item }) => item);
+    });
```

Comment:
It might be better to add some metadata to the cells themselves indicating their original row index, but this works quite well for now.